server.go

```diff
@@ -490,14 +490,15 @@ func NewFromConfig(logger *logrus.Logger, conf Config) (*Server, error) {
 		} else {
 			logger.Info("Successfully created AWS session")
 			svc = s3.New(sess)
-
-			plugin := &s3p.S3Plugin{
-				Logger:   log,
-				Svc:      svc,
-				S3Bucket: conf.AwsS3Bucket,
-				Hostname: ret.Hostname,
+			if conf.AwsS3Bucket != "" {
```

Comment:
Can we move this up to the check for the length of the AWS secrets? Logically they're related, so we should group them.